ddl/column.go

```diff
@@ -509,6 +546,11 @@ func onDropColumn(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 		// public -> write only
 		job.SchemaState = model.StateWriteOnly
 		colInfo.State = model.StateWriteOnly
+		if len(idxInfos) > 0 {
+			for _, indexInfo := range idxInfos {
+				indexInfo.State = model.StateWriteOnly
+			}
```

Comment:
Why not update hidden columns here, just like `setIndicesState(tblInfo, idxInfos, model.StateWriteOnly)`?

Reply:
This code is reference from `onDropIndex` function. In that function when `indexInfo.State` is `model.Statepublic`, there's no call on `updateHiddenColumns`. So I follow this logic here.

Reply:
Is that expected? @wjhuang2016